src/Either.ts

```diff
@@ -169,6 +169,13 @@ export function tryCatch<E, A>(f: Lazy<A>, onError: (e: unknown) => E): Either<E
   }
 }
 
+// Copied from https://github.com/Microsoft/TypeScript/issues/1897#issuecomment-338650717
+type AnyJson = boolean | number | string | null | JsonArray | JsonMap
+interface JsonMap {
```

Comment:
I initially use `type` but recursive type aliases are only supported in newer TS versions, so had to revert to `interface`.